Content/Items/Vitric/Weapons.PhoenixStorm.cs

```diff
@@ -0,0 +1,387 @@
+			alreadyHit.Add(target);
+			target.AddBuff(BuffID.OnFire, 240);
+			Projectile.penetrate++;
+		}
+
+		public override bool PreDraw(ref Color lightColor)
+		{
+			Texture2D tex = ModContent.Request<Texture2D>(Texture).Value;
+			Texture2D whiteTex = Request<Texture2D>(Texture + "_White").Value;
+			Texture2D glowTex = Request<Texture2D>(Texture + "_Glow").Value;
+			int frameHeight = tex.Height / Main.projFrames[Projectile.type];
+			Rectangle frame = new Rectangle(0, frameHeight * Projectile.frame, tex.Width, frameHeight);
+			
+			SpriteEffects spriteEffects = SpriteEffects.FlipHorizontally;
+			float rotation = Projectile.rotation;
+
+			if (Projectile.velocity.X < 0)
+				spriteEffects = SpriteEffects.None;
+			
+			if (player.HasMinionAttackTargetNPC || swooping)
+			{
+				int glowFrameHeight = glowTex.Height / Main.projFrames[Projectile.type];
+				Rectangle glowFrame = new Rectangle(0, glowFrameHeight * Projectile.frame, glowTex.Width, glowFrameHeight);
+				Color glowColor = Color.OrangeRed * glowOpacity;
+				glowColor.A = 0;
+				Main.spriteBatch.Draw(glowTex, Projectile.Center - Main.screenPosition, glowFrame, glowColor, rotation, glowFrame.Size() / 2, Projectile.scale * 1.2f, spriteEffects, 0f);
+			}
+
+			if (swooping)
+			{
+				Main.spriteBatch.End();
+				DrawTrail();
+				Main.spriteBatch.Begin(default, BlendState.Additive, default, default, default, default, Main.GameViewMatrix.TransformationMatrix);
+				for (int i = 0; i < oldPos.Count(); i++)
+				{
+					float opacity = i / (float)oldPos.Count();
+					Color color = Color.Lerp(Color.Orange, Color.Red, opacity);
+					Main.spriteBatch.Draw(tex, oldPos[i] - Main.screenPosition, frame, color * opacity, oldRot[i], frame.Size() / 2, Projectile.scale, spriteEffects, 0f);
+				}
+
+				Main.spriteBatch.End();
+				Main.spriteBatch.Begin(default, default, default, default, default, default, Main.GameViewMatrix.TransformationMatrix);
+			}
+			Main.spriteBatch.Draw(tex, Projectile.Center - Main.screenPosition, frame, lightColor, rotation, frame.Size() / 2, Projectile.scale, spriteEffects, 0f);
+			return false;
+		}
+
+		private void ManageCaches()
+		{
+			if (cache == null)
+			{
+				cache = new List<Vector2>();
+				for (int i = 0; i < 10; i++)
+				{
+					cache.Add(Projectile.Center);
+				}
+			}
+
+			cache.Add(Projectile.Center);
+
+			while (cache.Count > 10)
+			{
+				cache.RemoveAt(0);
+			}
+		}
+
+		private void ManageTrail()
+		{
+			trail ??= new Trail(Main.instance.GraphicsDevice, 10, new TriangularTip(4), factor => 13 * factor, factor =>
+			{
+				return Color.OrangeRed;
+			});
+
+			trail.Positions = cache.ToArray();
+			trail.NextPosition = Projectile.Center + Projectile.velocity;
+		}
+
+		public override bool MinionContactDamage() => swooping;
+
+		private void Swoop()
```

Comment:
This is placed a bit strangely around all of the trail drawing methods, you might want to move this up with the other AI subroutines